### ISSUE TITLE
How to have MDL aromaticity perception when doing smirks matching?

### ISSUE THREAD

**Issue by hjuinj** [rdkit/rdkit#1622]
- RDKit Version: 2017.03.3
- Platform: Centos7

I am attempting to use rdkit to implement support for the new SMIRNOFF format of the [open forcefield initiative](https://github.com/open-forcefield-group/openforcefield). This aims to allow automated parameterisation of small molecules for MD, as John Chodera showcased at the UGM last month (@davidlmobley @jchodera).

An important step in SMIRNOFF is to perform substructure matching of a molecule to different chemical environments encoded in smirks. The issue here is we found due to difference in aromaticity models used, the smirks matching can disagree between openeye and rdkit mol. We would like to adhere to the [MDL aromaticity model](https://docs.eyesopen.com/toolkits/python/oechemtk/aromaticity.html) which is what SMIRNOFF currently uses via openeye implementation.

As an example, the attached zipped file([DrugBank_1800.tar.gz](https://github.com/rdkit/rdkit/files/1416079/DrugBank_1800.tar.gz)) contains one pickled triazole molecule which I created by reading it in from .mol2 (see bottom) via openeye, then using a function I defined [here](https://github.com/open-forcefield-group/openforcefield/blob/swang/openforcefield/utils/utils.py#L498) to convert the openeye molecule into an equivalent rdkit molecule (this is to avoid some problems we saw with the molecule readers  #1590 #1596). 

With the default rdkit aromaticity model, the triazole satisfy the 4N + 2 rule and thus is regarded as aromatic. If I do the following:

```python
import pickle
from rdkit import Chem

rdkmol = pickle.load( open( "./1800.pkl", "rb" ) )
print(rdkmol.GetSubstructMatches(Chem.MolFromSmarts("[#7:1]:[#7:2]")))
```

I will get a match. However, the is **not** the case with the current SMIRNOFF openeye implementation, which regards the triazole ring as non-aromatic and hence no match.

So my question is essentially what is the easiest way for me to employ the MDL aromaticity model when doing the smirks matching?

This is most likely the last bit of difference between the oe and rdkit implementation that needs to be ironed out before getting a first release of the rdkit version of SMIRNOFF.

```
@<TRIPOS>MOLECULE
DrugBank_1800
   14    14     0     0     0
SMALL
NO_CHARGES

@<TRIPOS>ATOM
      1 C3         -0.1027   -3.9043    0.6923 C.ar      1 <0>         0.0000
      2 C2         -1.4525   -2.3475    1.0264 C.ar      1 <0>         0.0000
      3 C1         -0.0791   -0.4642    0.1125 C.cat     1 <0>         0.0000
      4 N4         -1.3457   -3.6451    1.1946 N.ar      1 <0>         0.0000
      5 N5          0.5664   -2.8487    0.2239 N.ar      1 <0>         0.0000
      6 N3         -0.3189   -1.8660    0.4494 N.pl3     1 <0>         0.0000
      7 N1         -1.0472    0.5017    0.3777 N.pl3     1 <0>         0.0000
      8 N2          1.1200   -0.0753   -0.4801 N.pl3     1 <0>         0.0000
      9 H6          0.3068   -4.9044    0.6777 H         1 <0>         0.0000
     10 H5         -2.2708   -1.6958    1.2811 H         1 <0>         0.0000
     11 H1         -1.9355    0.2694    0.8147 H         1 <0>         0.0000
     12 H2         -0.8831    1.4801    0.1526 H         1 <0>         0.0000
     13 H3          1.8484   -0.7569   -0.6800 H         1 <0>         0.0000
     14 H4          1.3153    0.8943   -0.7170 H         1 <0>         0.0000
@<TRIPOS>BOND
     1    1    4 1
     2    1    5 2
     3    2    4 2
     4    2    6 1
     5    3    6 1
     6    3    7 1
     7    3    8 2
     8    5    6 1
     9    1    9 1
    10    2   10 1
    11    7   11 1
    12    7   12 1
    13    8   13 1
    14    8   14 1
```

**Comment by davidlmobley** [comment]
Just to follow up on this, this is very important to us. Shuzhe has been putting a great deal of development time into getting SMIRNOFF to work with RDKit, and this is one of the few remaining bottlenecks. I think he, or someone else connected with the initiative, can probably get it to work without *too* much more effort but we need some pointers. Can we get some guidance on how we would go about implementing a custom aromaticity model or something similar to deal with this case?

Thanks.

**Comment by greglandrum** [comment]
Adding a custom aromaticity function is pretty easy as long as we can agree on what that model should be. From the table on the OE page cited above, it looks like only 6 rings are considered and that exocyclic double bonds don't have an impact.
Any other contraints that you're aware of?

**Comment by bannanc** [comment]
I'm mostly commenting so I'm involved in future discussion. As far as I remember, the MDL aromaticity model mostly changes 5-member heteroaromatic rings, possibly only in fused rings. I'm doing a little more research into the model to figure out if there are other exceptions.

**Comment by greglandrum** [comment]
Closed by #1693
